### DISCUSSION TITLE
DialogueLabel with history

### DISCUSSION THREAD

**Discussion by oninoshiko** [nathanhoad/godot_dialogue_manager#481] [Q&A] [answered]
My plan is to have a dedicated dialog section, but I would like it to show the history of what has been recently said. Is this already a feature I'm missing, or is there anything thoughts on how this could be implemented?

**Comment by oninoshiko** [comment, marked as answer]
Nevermind. I was able to transfer the text into a RichTextLabel using append text to get what I needed. If anyone else is looking to do this, here's what I did: 

```
func _on_finished_typing() -> void:
	%RichTextLabel.push_paragraph(HORIZONTAL_ALIGNMENT_LEFT)
	%RichTextLabel.append_text(%DialogueLabel.text)
	%RichTextLabel.pop()
	
	var para:int = %RichTextLabel.get_paragraph_count()
	if para > 50:
		%RichTextLabel.remove_paragraph(para - 1)
	
	%DialogueLabel.text=""
```

**Reply by dannyparisi** [reply]
Hello, sorry to necro but I tried using your exact code and it's still overwriting the previous text. I'm pulling my hair out trying to make this work. Any chance you had any more detail about how you got it working?
